src/Adapter/Address/AbstractAddressHandler.php

```diff
@@ -77,4 +78,20 @@ protected function deleteAddress(Address $address): bool
             throw new AddressException(sprintf('An error occurred when deleting Address object with id "%s".', $address->id));
         }
     }
+
+    /**
+     * @param Address $address
+     *
+     * @throws InvalidAddressFieldException
+     * @throws PrestaShopException
+     */
+    protected function validateAddress(Address $address): void
+    {
+        if (true !== ($validateResult = $address->validateFields(false, true))) {
+            throw new InvalidAddressFieldException(sprintf('Address contains invalid field values: %s', $validateResult));
+        }
+        if (true !== ($validateResult = $address->validateFieldsLang(false, true))) {
+            throw new InvalidAddressFieldException(sprintf('Address contains invalid field values: %s', $validateResult));
+        }
+    }
```

Comment:
```suggestion
        if (true !== ($validateResult = $address->validateFields(false, true)) 
            || true !== ($validateResult = $address->validateFieldsLang(false, true))
        ) {
            throw new InvalidAddressFieldException(sprintf('Address contains invalid field values: %s', $validateResult));
        }
    }
```

Compute both, do not duplicate throw